src/hotspot/share/gc/g1/g1ConcurrentRefineThread.cpp

```diff
@@ -79,6 +79,7 @@ void G1ConcurrentRefineThread::run_service() {
     } else {
       _vtime_accum = 0.0;
     }
+    maybe_update_threads_cpu_time();
```

Comment:
I think the lines above here: 
```
if (os::supports_vtime()) {
  _vtime_accum = (os::elapsedVTime() - _vtime_start);
} else {
  _vtime_accum = 0.0;
}
```
Should be extracted out into the new method and instead of calling it `maybe_update_threads_cpu_time()` just call `track_usage()` or `track_cpu_time()`. The the implementation in the primary thread can then call this and do the extra tracking.